src/structures/gdt.rs

```diff
@@ -112,17 +112,19 @@ impl GlobalDescriptorTable {
     /// * The user must make sure that the entries are well formed
     /// * The provided slice **must not be larger than 8 items** (only up to the first 8 will be observed.)
     #[inline]
-    #[cfg(feature = "const_fn")]
     pub const unsafe fn from_raw_slice(slice: &[u64]) -> GlobalDescriptorTable {
-        assert!(
-            slice.len() <= 8,
-            "initializing a GDT from a slice requires it to be **at most** 8 elements."
-        );
         let next_free = slice.len();
-
         let mut table = [0; 8];
         let mut idx = 0;
 
+        #[cfg(feature = "const_fn")]
+        assert!(
+            next_free <= 8,
+            "initializing a GDT from a slice requires it to be **at most** 8 elements."
+        );
+        #[cfg(not(feature = "const_fn"))]
+        table[next_free]; // Will fail if slice.len() > 8
+
```

Comment:
Why can't we use `assert!` in every case? Also, `table[next_free]` panics if `slice.len() >= 8`, not `slice.len() > 8`.

Reply:
`assert!` doesn't work as const panics are not yet stable. See: https://github.com/rust-lang/rust/issues/85194

Good pont on the off-by-one error though, i'll fix it

Reply:
@josephlr Did we end up fixing this yet?

Reply:
Whoops, I forgot about this. PR for fix: https://github.com/rust-osdev/x86_64/pull/269